eq_translations/survey_schema.py

```diff
@@ -1,245 +1,204 @@
+                    yield self._get_pointer_dict(pointer, schema_element, with_context)
 
-        list_pointers = find_pointers_to(self.schema, "list")
-        for list_pointer in list_pointers:
-            schema_element = resolve_pointer(self.schema, list_pointer)
-            if isinstance(schema_element, list):
-                for i, p in enumerate(schema_element):
-                    # placeholders are being skipped as they are dealt with elsewhere
-                    if not is_placeholder(p):
-                        pointers.append(f"{list_pointer}/{i}")
+    def _get_pointers_from_dict_element(self, pointer, schema_element, with_context):
+        plural_forms = schema_element.get("text_plural")
+        if plural_forms:
+            new_pointer = f"{pointer}/text_plural"
+            yield self._get_pointer_dict(new_pointer, plural_forms, with_context)
 
-        return pointers
+        for element in schema_element:
+            value = schema_element[element]
+            if isinstance(value, str):
+                new_pointer = f"{pointer}/{element}"
+                yield self._get_pointer_dict(new_pointer, value, with_context)
 
-    def get_title_pointers(self):
-        """
-        Titles need to be handled separately as they may require context for translation
-        """
-        return find_pointers_to(self.schema, "title")
+    def _get_pointers_from_list_element(self, pointer, schema_element, with_context):
+        for index, _ in enumerate(schema_element):
+            value = schema_element[index]
+            if isinstance(value, str):
+                new_pointer = f"{pointer}/{index}"
+                yield self._get_pointer_dict(new_pointer, value, with_context)
 
-    def get_answer_pointers(self):
+    def _get_pointer_dict(self, pointer, value=None, with_context=False):
         """
-        Labels for options/answers need to be handled separately as they require context
+        :param pointer: Defines a string syntax for identifying a specific value within the schema.
+        :param value: The resolved value of the pointer. This is a Tuple for plural forms, and a String for all other elements
+        :param with_context: Boolean flag to signify whether this pointer should have the question title as context.
+        :return: A dictionary.
         """
-        return find_pointers_to(self.schema, "label")
-
-    def get_parent_id(self, pointer):
-        resolved_data = resolve_pointer(self.schema, pointer)
+        pointer_dict = {"pointer": pointer, "value": value}
 
-        if isinstance(resolved_data, dict) and "id" in resolved_data:
-            return resolved_data["id"]
+        if with_context and f"/answers/" in pointer:
+            message_context = self._get_message_context_from_pointer(pointer)
+            pointer_dict["context"] = message_context
 
-        parent_pointer = get_parent_pointer(pointer)
-        return self.get_parent_id(parent_pointer)
+        return pointer_dict
```

Comment:
If we're creating an object based on parameters passed on creation, I think we should consider creating a `TranslatableItem` class. On init it could set the appropriate properties and context. It would also help document what a translatable item is.